auth-lib/src/main/java/com/spotify/sdk/android/auth/app/SpotifyNativeAuthUtil.java

```diff
@@ -99,46 +106,78 @@ public boolean startAuthActivity() {
+            intent = tryResolveActivity(context,
+                    SPOTIFY_PACKAGE_NAME + suffix,
+                    sha1HashUtil);
             if (intent != null) {
                 break;
             }
         }
         return intent;
     }
 
-    private Intent tryResolveActivity(String packageName) {
+    /**
+     * Check if a version of the Spotify main application is installed
+     *
+     * @param context The context of the caller, used to check if the app is installed
+     * @return True if a Spotify app is installed, false otherwise
+     */
+    public static boolean isSpotifyInstalled(@NonNull Context context) {
+        return isSpotifyInstalled(context, new Sha1HashUtil());
+    }
+
+    @VisibleForTesting
+    static boolean isSpotifyInstalled(@NonNull Context context, @NonNull Sha1HashUtil sha1HashUtil) {
+        return createAuthActivityIntent(context, sha1HashUtil) != null;
+    }
 
+    @Nullable
+    private static Intent tryResolveActivity(@NonNull Context context,
+                                             @NonNull String packageName,
+                                             @NonNull Sha1HashUtil sha1HashUtil) {
         Intent intent = new Intent(SPOTIFY_AUTH_ACTIVITY_ACTION);
         intent.setPackage(packageName);
 
-        ComponentName componentName = intent.resolveActivity(mContextActivity.getPackageManager());
+        ComponentName componentName = intent.resolveActivity(context.getPackageManager());
 
         if (componentName == null) {
             return null;
         }
 
-        if (!validateSignature(componentName.getPackageName())) {
+        if (!validateSignature(context, componentName.getPackageName(), sha1HashUtil)) {
             return null;
         }
 
         return intent;
     }
 
     @SuppressLint("PackageManagerGetSignatures")
-    private boolean validateSignature(String spotifyPackageName) {
+    private static boolean validateSignature(@NonNull Context context,
+                                             String spotifyPackageName,
+                                             @NonNull Sha1HashUtil sha1HashUtil) {
         try {
-            final PackageInfo packageInfo = mContextActivity.getPackageManager().getPackageInfo(spotifyPackageName, PackageManager.GET_SIGNATURES);
+            final PackageInfo packageInfo = context.getPackageManager().getPackageInfo(spotifyPackageName, PackageManager.GET_SIGNATURES);
             if (packageInfo.signatures == null) {
                 return false;
             }
 
             for (Signature actualApkSignature : packageInfo.signatures) {
                 final String signatureString = actualApkSignature.toCharsString();
-                final String sha1Signature = sha1Hash(signatureString);
+                final String sha1Signature = sha1HashUtil.sha1Hash(signatureString);
                 for (String knownSpotifyHash : SPOTIFY_SIGNATURE_HASH) {
                     if (knownSpotifyHash.equals(sha1Signature)) {
```

Comment:
This code is comparing a base16 hash, where the alphabet is case-insensitive. Consider performing a case-insensitive comparison to avoid false negatives. It is a best practice to perform time constant hash comparisons using, for example, `MessageDigest.isEqual`. If time constant comparison is performed, this can be done at the byte level before base16 encoding is performed.